lib/dialects/postgres/query-generator.js

```diff
@@ -254,7 +254,9 @@ const QueryGenerator = {
   removeColumnQuery(tableName, attributeName) {
     const quotedTableName = this.quoteTable(this.extractTableDetails(tableName));
     const quotedAttributeName = this.quoteIdentifier(attributeName);
-    return `ALTER TABLE ${quotedTableName} DROP COLUMN ${quotedAttributeName};`;
+    let query = `ALTER TABLE ${quotedTableName} DROP COLUMN ${quotedAttributeName};`;
+    query += this.pgEnumDrop(tableName, attributeName);
```

Comment:
I would add a `RESTRICT` here, just to be explicitly on the safe side (to prevent dropping the enum if other types depend on it) https://www.postgresql.org/docs/8.2/static/sql-droptype.html